hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieMultiTableDeltaStreamer.java

```diff
@@ -370,50 +441,124 @@ public static void main(String[] args) throws IOException {
+      if (!failedTables.isEmpty()) {
+        logger.info("Ingestion failed for topics: " + failedTables);
+        break;
+      }
+      successTables.clear();
+    } while (isContinuousMode);
   }
 
   public static class Constants {
+    // When there are multiple sources, you can use this configuration item to set an independent checkpoint for the source.
+    public static final String SOURCE_CHECKPOINT = "hoodie.deltastreamer.current.source.checkpoint";
+
+    // If there are multiple sources, you can use this configuration item to set an alias for the source to distinguish the source.
+    // In addition, the alias is used as a suffix to distinguish the CHECKPOINT_KEY and CHECKPOINT_RESET_KEY of each source.
+    public static final String SOURCE_NAME = "hoodie.deltastreamer.current.source.name";
+
     public static final String KAFKA_TOPIC_PROP = "hoodie.deltastreamer.source.kafka.topic";
+
     private static final String SOURCE_SCHEMA_REGISTRY_URL_PROP = "hoodie.deltastreamer.schemaprovider.registry.url";
+
     private static final String TARGET_SCHEMA_REGISTRY_URL_PROP = "hoodie.deltastreamer.schemaprovider.registry.targetUrl";
+
     public static final String HIVE_SYNC_TABLE_PROP = "hoodie.datasource.hive_sync.table";
+
     private static final String SCHEMA_REGISTRY_BASE_URL_PROP = "hoodie.deltastreamer.schemaprovider.registry.baseUrl";
+
     private static final String SCHEMA_REGISTRY_URL_SUFFIX_PROP = "hoodie.deltastreamer.schemaprovider.registry.urlSuffix";
+
     private static final String SCHEMA_REGISTRY_SOURCE_URL_SUFFIX = "hoodie.deltastreamer.schemaprovider.registry.sourceUrlSuffix";
+
     private static final String SCHEMA_REGISTRY_TARGET_URL_SUFFIX = "hoodie.deltastreamer.schemaprovider.registry.targetUrlSuffix";
+
     private static final String TABLES_TO_BE_INGESTED_PROP = "hoodie.deltastreamer.ingestion.tablesToBeIngested";
+
+    // This configuration item specifies the database name and table name of the source. The format is "database.table".
+    // It is recommended that table name be the same as the alias of the source. If there are multiple sources, separate them with commas.
+    public static final String SOURCES_TO_BE_BOUND = "hoodie.deltastreamer.source.sourcesToBeBound";
+
+    private static final String SOURCE_PREFIX = "hoodie.deltastreamer.source.";
+
     private static final String INGESTION_PREFIX = "hoodie.deltastreamer.ingestion.";
+
     private static final String INGESTION_CONFIG_SUFFIX = ".configFile";
+
     private static final String DEFAULT_CONFIG_FILE_NAME_SUFFIX = "_config.properties";
+
     private static final String TARGET_BASE_PATH_PROP = "hoodie.deltastreamer.ingestion.targetBasePath";
+
     private static final String LOCAL_SPARK_MASTER = "local[2]";
-    private static final String FILE_DELIMITER = "/";
-    private static final String DELIMITER = ".";
+
+    public static final String PATH_SEPARATOR = "/";
+
+    public static final String PATH_CUR_DIR = ".";
```

Comment:
I would prefer to keep this as "DELIMITER".

Reply:
+1

Reply:
Thank you for your advice. I have changed corresponding content in the code.